docs/onboarding/azure-devops-pipelines.md

```diff
@@ -171,39 +171,83 @@ Instructions:
 
-    **Sample environment YAML**
+    **Sample environment YAML (v0.9.0 or later)**
 
-    ```yaml
+    ```yml
     variables:
+    
+      # Management Groups
+      var-managementgroup-hierarchy: >
+        {
+          "name": "Tenant Root Group",
+          "id": "abcddfdb-bef5-46d9-99cf-ed67dabc8783",
+          "children": [
+            {
+              "name": "pubsec",
+              "id": "pubsec",
+              "children": [
+                {
+    ...
+    ```
+
+    In CanadaPubSecALZ v0.9.0 or later, the management group hierarchy can be defined in the environment configuration file. It is represented in JSON as a set of nested management group objects with the structure:
+
+    ```json
+    {
+      "name": "Management Group Display Name",
+      "id": "ManagementGroupIdentifier_NoSpaces",
+      "children": []
+    }
+    ```
 
+    Each `"children": []` element is an array of the same object type, and may contain zero or more child management group object definitions, which in turn may contain their own arrays of zero or more child management group object definitions.
+
+    Observe the following guidance when configuring your management group hierarchy:
+
+      * Specify the `id` and `name` of your existing Azure AD tenant for the topmost management group definition. The `id` attribute for this element is mapped into the legacy `var-parentManagementGroupId` pipeline variable for backward compatibility.
+      * Specify only 1 child management group definition for the topmost management group definition. You can specify more than 1 child of the topmost management group definition, but it is the first child of the topmost level that will be considered the root of of your management group hierarchy, and is the scope that the `policy-ci` pipeline will use to deploy built-in and custom policies. The `id` attribute for this element is mapped into the legacy `var-topLevelManagementGroupName` pipeline variable for backward compatibility.
+      * The `id` attribute for management group elements can only be an ASCII letter, digit, `-`, `_`, `(`, `)`, `.` and cannot end with a period. In the sample environment configuration file (`CanadaESLZ-main.yml`), we illustrate a convention that prepends the id of the parent management group to the id of each child management group. This is an example only and not a requirement. You are welcome to choose any management group id convention that best suits your needs.
+      * If you are using **CanadaPubSecALZ v0.9.0 or later** and **do not** include a `var-managementgroup-hierarchy` variable setting in your configuration, it will fallback to using the legacy pipeline variables `var-parentManagementGroupId` and `var-topLevelManagementGroupName`. This is to ensure backward compatibility, enabling newer versions of the code to run with older environment configurations.
```

Comment:
Instead of referencing the parameters as `legacy` pipeline variables, what if we say that it will fall back to hardcoded values?

Reply:
Removed legacy